### PR TITLE
DOC: extend "building reproducible binaries" page

### PR DESCRIPTION
**Pull request by rgommers**
Add details on:

1. Build and runtime dependencies
2. How to strip the test suite (follows up on gh-20712)

### CODE REVIEW

File: doc/source/building/redistributable_binaries.rst [hunk 1]
```diff
@@ -1,6 +1,10 @@
 Building redistributable binaries
 =================================
 
+*This intended audience for this section is anyone who wants to build SciPy and
```
Comment (tylerjereddy) [inline doc/source/building/redistributable_binaries.rst:4]:
```suggestion
*The intended audience for this section is anyone who wants to build SciPy and
```

File: doc/source/building/redistributable_binaries.rst [hunk 2]
```diff
@@ -10,12 +14,90 @@ artifacts" <https://pypackaging-native.github.io/meta-topics/build_steps_concept
...
+    #   3. The <2.3 upper bound is for matching the numpy deprecation policy,
+    #      it should not be loosened.
+    "numpy>=2.0.0rc1,<2.3",
+
+Non-Python build requirements are:
+
+- C, C++ and Fortran compilers
+- BLAS and LAPACK libraries
+- ``ninja``
+- ``pkg-config``
```
Comment (tylerjereddy) [inline doc/source/building/redistributable_binaries.rst:50]: Can you get away with the CMake equivalent? I thought I saw the config stage checking for that at one point when I didn't have `pkg-config` for finding the linalg backend, but yeah I always use `pkg-config` now I think.

File: doc/source/building/redistributable_binaries.rst [hunk 3]
```diff
@@ -10,12 +14,90 @@ artifacts" <https://pypackaging-native.github.io/meta-topics/build_steps_concept
...
+
+    $ python -m build -wnx -Cinstall-args=--tags=runtime,python-runtime,devel
+
+.. note::
+
+   Note that in the above command ``-wnx`` means ``--wheel --no-isolation
+   --skip-dependency-check``. It assumes that the packager has already set up
+   the build environment, which is usually the case for distro packaging. The
+   install tags feature works equally well with isolated builds (e.g. ``pip
+   install numpy --no-binary -Cinstall-args=--tags=runtime,python-runtime,devel``).
```
Comment (tylerjereddy) [inline doc/source/building/redistributable_binaries.rst:78]:
```suggestion
   install scipy --no-binary -Cinstall-args=--tags=runtime,python-runtime,devel``).
```

**Review by tylerjereddy — COMMENTED** [review]
Seems to read well enough, I only made tiny suggestions.

By the way, is the plan for me to have the `1.14.0` wheels separated in this way? Are we already ready for that?

**Comment by rgommers** [comment]
> By the way, is the plan for me to have the `1.14.0` wheels separated in this way? Are we already ready for that?

No, I wasn't intending that. I think to do that we need to implement support in wheel build CI. And it requires some care in adding things like version checks to ensure that a test suite can only run against an identical version of the package, the `scipy.test()` error message is clear, etc. I hope we can do this for 1.15.0. For now it's already useful for conda-forge, Pyodide and the AWS Lambda packager.

File: doc/source/building/redistributable_binaries.rst (hunk 2 above)
Comment (rgommers) [reply doc/source/building/redistributable_binaries.rst:50]: You can in principle, but it's not well-tested so I would not want to go into that here.

**Comment by tylerjereddy** [comment]
Ok, the rendered version of the docs seems good enough locally to merge.

There may be one more blemish we could clean up later, removing an extra `$` perhaps:

image

**Pull request merged**